src/lambda-calculus.js

```diff
@@ -417,4 +428,4 @@ function printStackTrace(error, term, stack) {
     console.error( stack.slice(stackCutoff).reverse().map( v => `\twhile evaluating ${ v }`).join('\n') );
 }
 
-Object.defineProperty( Function.prototype, "valueOf", { value: function valueOf() { return toInt(this); } } );
+Object.defineProperty( Function.prototype, "valueOf", { value: function valueOf() { if (this.term) return toInt(this); } } );
```

Comment:
else return this;

`valueOf` should always return some equivalent of its this.